tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -732,13 +732,19 @@ def __init__(self, data, manager=None):
     def __call__(self, i):
         if self.cached_costs is None:
             self.cached_costs = self._query_costs()
-        id = i['id'].lower()
-        if id not in self.cached_costs:
+        id = i['id'].lower() + "/"
+
+        costs_rg = self.cached_costs.get(ResourceIdParser.get_resource_group(id), {})
+        costs = [costs_rg[k] for k in costs_rg.keys() if (k + '/').startswith(id)]
+
+        if not costs:
             return False
 
-        cost = self.cached_costs[id]
-        i['c7n:cost'] = cost
-        result = super(CostFilter, self).__call__(cost)
+        total_cost = costs[0]
+        total_cost['PreTaxCost'] = sum(c['PreTaxCost'] for c in costs)
```

Comment:
Since 'PreTaxCost' is being hardcoded as the key from the schema - is the idea in the future for the cost to be calculated by a key provided (PreTaxCost vs Currency)? In which case should the 'key' be used here instead of hardcoding 'PreTaxCost' ?

Flipside - since the key comes back in the schema but the user can't set it right now - should we just remove the key from the schema?

Reply:
Removed key from schema for now. I think it makes sense to have a separate filter for currency.. What do you think?